src/aosm/azext_aosm/tests/latest/unit_test/test_helm_chart_processor.py

```diff
@@ -0,0 +1,35 @@
+from unittest import TestCase
+from azext_aosm.tests.latest.tests_utils import update_input_file, get_tests_path
+from azext_aosm.cli_handlers.onboarding_cnf_handler import (
+    OnboardingCNFCLIHandler,
+)
+from pathlib import Path
+
+CNF_NF_AGENT_INPUT_TEMPLATE_NAME = "input-nf-agent-cnf-template.jsonc"
+CNF_NF_AGENT_INPUT_FILE_NAME = "test_helm_chart_processor_input-nf-agent-cnf.jsonc"
+
+
+class TestHelmChartProcessor(TestCase):
+    def setUp(self):
+        config_file = update_input_file(
+            CNF_NF_AGENT_INPUT_TEMPLATE_NAME,
+            CNF_NF_AGENT_INPUT_FILE_NAME,
+            params={
+                "tests_directory": get_tests_path(),
+            },
+        )
+
+        handler = OnboardingCNFCLIHandler(Path(config_file))
+        # We want to test a specific private method so disable the pylint warning
+        # pylint: disable=protected-access
+        handler._validate_helm_template()
+        self.helm_chart_processor = handler.processors[0]
```

Comment:
Instantiating all these objects which aren't objects under test makes these units tests less 'unit-y' and more integration-like. 

Right now I don't think they're awful:
1. This is probably easier than mocking
2. We do get a bit of cheeky extra coverage

However:
1. If the test fails in the setup, it's not immediately obvious where the problem is (which is a key advantage of unit tests)
2. We shouldn't allow this to creep further. If we find we're instantiating more and more objects in our unit tests, that's a warning sign that we're doing them wrong.

I'm OK to leave these as is for now, unless you particularly want to mock things out, in which case go for it.